integrations/azure_ai_search/src/haystack_integrations/document_stores/azure_ai_search/document_store.py

```diff
@@ -0,0 +1,450 @@
+
+        For more information on parameters, see the [official Azure AI Search documentation](https://learn.microsoft.com/en-us/azure/search/)
+        """
+
+        azure_endpoint = azure_endpoint or os.environ.get("AZURE_SEARCH_SERVICE_ENDPOINT") or None
+        if not azure_endpoint:
+            msg = "Please provide an Azure endpoint or set the environment variable AZURE_OPENAI_ENDPOINT."
+            raise ValueError(msg)
+
+        api_key = api_key or os.environ.get("AZURE_SEARCH_API_KEY") or None
+
+        self._client = None
+        self._index_client = None
+        self._index_fields = []  # type: List[Any]  # stores all fields in the final schema of index
+        self._api_key = api_key
+        self._azure_endpoint = azure_endpoint
+        self._index_name = index_name
+        self._embedding_dimension = embedding_dimension
+        self._dummy_vector = [-10.0] * self._embedding_dimension
+        self._metadata_fields = metadata_fields
+        self._vector_search_configuration = vector_search_configuration or DEFAULT_VECTOR_SEARCH
+        self._create_index = create_index
+        self._kwargs = kwargs
+
+    @property
+    def client(self) -> SearchClient:
+
+        # resolve secrets for authentication
+        resolved_endpoint = (
+            self._azure_endpoint.resolve_value() if isinstance(self._azure_endpoint, Secret) else self._azure_endpoint
+        )
+        resolved_key = self._api_key.resolve_value() if isinstance(self._api_key, Secret) else self._api_key
+
+        credential = AzureKeyCredential(resolved_key) if resolved_key else DefaultAzureCredential()
+        try:
+            if not self._index_client:
+                self._index_client = SearchIndexClient(resolved_endpoint, credential, **self._kwargs)
+            if not self.index_exists(self._index_name):
+                # Create a new index if it does not exist
+                logger.debug(
+                    "The index '%s' does not exist. A new index will be created.",
+                    self._index_name,
+                )
+                self.create_index(self._index_name)
+        except (HttpResponseError, ClientAuthenticationError) as error:
+            msg = f"Failed to authenticate with Azure Search: {error}"
+            raise AzureAISearchDocumentStoreConfigError(msg) from error
+
+        if self._index_client:  # type: ignore # self._index_client is not None (verified in the run method)
+            # Get the search client, if index client is initialized
+            index_fields = self._index_client.get_index(self._index_name).fields
+            self._index_fields = [field.name for field in index_fields]
+            self._client = self._index_client.get_search_client(self._index_name)
+        else:
+            msg = "Search Index Client is not initialized."
+            raise AzureAISearchDocumentStoreConfigError(msg)
+
+        return self._client
+
+    def create_index(self, index_name: str, **kwargs) -> None:
```

Comment:
```suggestion
    def _create_index(self, index_name: str, **kwargs) -> None:
```
This seems an internal method, no?